src/charm.py

```diff
@@ -1,29 +1,130 @@
 #!/usr/bin/env python3
 # Copyright 2021 Canonical Ltd.
 # See LICENSE file for licensing details.
+import json
```

Comment:
Do you really want all this code in you charm.py ? Alternatively does it make sense to factor it out into a module (say kubernetes_service.py) of its own.

Reply:
This isn't a bad idea, given that other folk in the O11y team will need this too. I don't think for now this should be a charm library because we will ultimately fix this in Juju, but worth pulling out into its own file with it's own tests probably.

Reply:
Done